mlflow/tracking/client.py

```diff
@@ -910,6 +914,50 @@ def log_artifacts(self, run_id, local_dir, artifact_path=None):
         """
         self._tracking_client.log_artifacts(run_id, local_dir, artifact_path)
 
+    @contextlib.contextmanager
+    def _log_artifact_helper(self, run_id, artifact_file):
+        """
+        Yields a temporary path to store a file, and then calls `log_artifact` against that path.
+        """
+        norm_path = posixpath.normpath(artifact_file)
+        filename = posixpath.basename(norm_path)
+        artifact_path = posixpath.dirname(norm_path)
```

Comment:
Used `normpath` here to handle a relative path like `"./a.txt"` or `"a/../b.txt"`.

Reply:
Could you add a test for logging to such paths?

Reply:
done!